### ISSUE TITLE
Denoland deploy

### ISSUE THREAD

**Issue by fabiancook** [kenchris/urlpattern-polyfill#52]
Would it be possible to have this added to the denoland registry?

This would involve adding a webhook to this module, which would then publish the typescript source. 

Details are listed here: https://deno.land/x

**Comment by fabiancook** [comment]
Using an import map, this polyfill can be utilised directly through GitHub:

```json
{
  "imports": {
    "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern.ts",
    "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-utils": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-utils.ts",
    "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern.interfaces": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern.interfaces.ts",
    "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/path-to-regex-modified": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/path-to-regex-modified.ts",
    "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern-parser": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/url-pattern-parser.ts",
    "urlpattern-polyfill": "https://raw.githubusercontent.com/kenchris/urlpattern-polyfill/main/src/index.ts",
  }
}
```

This would be similar if it was deployed to deno.land unless these typescript files were process further to include extensions for deployment.

**Comment by fabiancook** [comment]
Ignoring the later failing jobs in these actions, we can see this code running across deno, node, webkit, chrome, and firefox:

https://github.com/virtualstate/app-history/blob/7171f6aea0e26847525ff65796b96447a52e4169/src/tests/examples/url-pattern.ts#L12-L21

```
appHistory.addEventListener("navigate", ({ destination }) => {
        const identifiedTest = new URLPattern({
            pathname: "/test/:id"
        });
        if (identifiedTest.test({ pathname: destination.url })) {
            body.innerHTML = destination.getState<{ innerHTML: string }>().innerHTML;
        } else {
            throw new Error(unexpectedPage);
        }
    });
```

Running the tests we can see the matcher is working:

```
START  urlPatternExample
Updated window pathname to /test/1
Updated window pathname to /test/2
PASS   urlPatternExample
```

- [Node Test](https://github.com/virtualstate/app-history/runs/4617551990?check_suite_focus=true#step:7:119) - Uses native node.js resolution + yarn/npm
- [Deno Test](https://github.com/virtualstate/app-history/runs/4617551990?check_suite_focus=true#step:8:137) - Uses the above import map
- [Chrome Test](https://github.com/virtualstate/app-history/runs/4617551990?check_suite_focus=true#step:7:246) - Uses skypack CDN variant of package) 
- [WebKit Test](https://github.com/virtualstate/app-history/runs/4617551990?check_suite_focus=true#step:7:389) - Uses rollup, no import map support
- [Firefox Test](https://github.com/virtualstate/app-history/runs/4617551990?check_suite_focus=true#step:7:521)  - Uses rollup, no import map support

**Comment by kenchris** [comment]
I don't know much about deno land or how to publish it there. Could you share instructions?

**Comment by fabiancook** [comment]
Now that your module that is published to npm with `exports`, it is also directly compatible with [skypack](https://www.skypack.dev/view/urlpattern-polyfill) ([maybe it was before too?](https://cdn.skypack.dev/urlpattern-polyfill@v1.0.0-rc1))

I found that I can now just use the import map:

```json
{
  "imports": {
    "urlpattern-polyfill": "https://cdn.skypack.dev/urlpattern-polyfill"
  }
}
```

-------------------

However some might still want to access from deno.land, here are steps to publish to the deno.land registry:

1. Go to [https://deno.land/x](https://deno.land/x)
2. Click "Publish Module"

![image](https://user-images.githubusercontent.com/4337080/149594876-bf9c83e3-cd89-444a-8eba-bc690af428e2.png)

3. Add a module name

![image](https://user-images.githubusercontent.com/4337080/149594950-f4d976ea-7acb-40e8-b71b-c12197d94e4e.png)

4. Add a subdirectory, `src/` here is usable because your source is typescript, deno can directly utilise it

![image](https://user-images.githubusercontent.com/4337080/149595038-4132244c-2420-401d-9a56-f359adeb3410.png)

5. Follow these steps to add the webhook to this repository:

![image](https://user-images.githubusercontent.com/4337080/149595322-af80cf4f-6ed8-4acf-9327-d1e235b8abc9.png)

I think the problem here would be though... the import map still will be a bit messy because of deno's requirement for extensions to be present for exports, while this isn't common practice when writing typescript

Once deployed the import map would probably look something like:

```json
{
  "imports": {
    "https://deno.land/x/urlpattern_polyfill/url-pattern": "https://deno.land/x/urlpattern_polyfill/url-pattern.ts",
    "https://deno.land/x/urlpattern_polyfill/url-utils": "https://deno.land/x/urlpattern_polyfill/url-utils.ts",
    "https://deno.land/x/urlpattern_polyfill/url-pattern.interfaces": "https://deno.land/x/urlpattern_polyfill/url-pattern.interfaces.ts",
    "https://deno.land/x/urlpattern_polyfill/path-to-regex-modified": "https://deno.land/x/urlpattern_polyfill/path-to-regex-modified.ts",
    "https://deno.land/x/urlpattern_polyfill/url-pattern-parser": "https://deno.land/x/urlpattern_polyfill/url-pattern-parser.ts",
    "urlpattern-polyfill": "https://deno.land/x/urlpattern_polyfill"
  }
}
```

Even then, I'm not sure how deno's resolution exactly works, and it might rely on the redirected to url for this individual mapping, meaning possibly this is the required import map with version pinning included:


```json
{
  "imports": {
    "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern.ts",
    "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-utils": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-utils.ts",
    "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern.interfaces": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern.interfaces.ts",
    "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/path-to-regex-modified": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/path-to-regex-modified.ts",
    "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern-parser": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2/url-pattern-parser.ts",
    "urlpattern-polyfill": "https://deno.land/x/urlpattern_polyfill@v1.0.0-rc2"
  }
}
```

--------------------

TLDR seems like skypack is a better goto here, no additional steps required there.

**Comment by kenchris** [comment]
Why do you need it in Deno? Deno supports URLPattern natively now

**Comment by fabiancook** [comment]
Ha! Well, I guess I could work a local polyfill file to detect globalThis.URLPattern exists!

I'm writing code that is compatible across all platforms, so using the polyfill is a must for others like node or the browser. I would have expected this polyfill to eventually provide the global where it is present.

**Comment by kenchris** [comment]
I see. That is tracked in https://github.com/kenchris/urlpattern-polyfill/issues/47 so closing this one
